app/models/test_passage.rb

```diff
@@ -0,0 +1,52 @@
+class TestPassage < ApplicationRecord
+  belongs_to :user
+  belongs_to :test
+  belongs_to :current_question, class_name: 'Question', optional: true
+
+  before_validation :before_validation_set_first_question, on: :create
+  after_validation :after_validation_set_next_question, on: :update
+
+  PASSING_SCORE = 85
+
+  def accept!(answer_ids)
+    if correct_answer?(answer_ids)
+      self.correct_questions += 1
+    end
+
+    save!
+  end
+
+  def completed?
+    current_question.nil?
+  end
+
+  def success_percent
+    (self.correct_questions.fdiv(test.questions.count)*100).round
+  end
+
+  def success?
+    if success_percent >= PASSING_SCORE
+      true
+    else
+      false
+    end
```

Comment:
```suggestion
    success_percent >= PASSING_SCORE
```

Нет необходимости в `if`.